### PR TITLE
Update address form

### PR DESCRIPTION
**Pull request by mgcamisa**
* Update address form placeholders and labels
* Get saved last address, not first address, so it is up to date with their most recent changes
* Add optional state choicefield for American states (template will be updated in TC)
* Update translations

TODO: update billing address form to use select field as well

### CODE REVIEW

File: vendor/models/choice.py [hunk 1]
```diff
@@ -307,3 +307,57 @@ class Country(models.IntegerChoices):
     ZA = 710, _("South Africa")
     ZM = 894, _("Zambia")
     ZW = 716, _("Zimbabwe")
+
+
+class StateChoices(models.TextChoices):
```
Comment (rhimmelbauer) [inline vendor/models/choice.py:312]: I would change the class name to be more specific to the USA. Don't know if `UnitedStatesStateChoices` or `USAStateChoices`

File: vendor/views/vendor.py [hunk 2]
```diff
@@ -89,12 +89,12 @@ def get(self, request, *args, **kwargs):
         invoice.status = InvoiceStatus.CHECKOUT
         invoice.save()
 
-        existing_account_address = Address.objects.filter(profile__user=request.user, profile__site=get_site_from_request(request))
+        existing_account_address = Address.objects.filter(profile__user=request.user, profile__site=get_site_from_request(request)).last()
```
Comment (rhimmelbauer) [inline vendor/views/vendor.py:92]: Nice!!!

**Review by rhimmelbauer — COMMENTED** [review]
Just a single note on the StateChocies Name. Going to run authorizenet test and approve afterwards so you can just change the name and merge :)

**Comment by rhimmelbauer** [comment]
Also don't forget to update version :)

**Comment by rhimmelbauer** [comment]
Unit test passed :) pr approved

**Review by rhimmelbauer — APPROVED** [review]
Merge when you have you StateChoice name update :)

**Comment by mgcamisa** [comment]
> Also don't forget to update version :)

version was already bumped in the previous pr, thanks!

**Pull request merged**